load_split_data.py

```diff
@@ -0,0 +1,165 @@
+        elif drug_inds[i] in drugs_validation:
+            if target_inds[i] in targets_training:
+                S3_validation_indices.append(i)
+            elif target_inds[i] in targets_validation:
+                S4_validation_indices.append(i)
+    
+    # Create a data frame that contains all the information of the created splits for saving purposes.
+    df_indices = pd.concat([pd.DataFrame({'roundID':random_seed, 'subset':'training', 'setting':'all', 'index':training_indices, 'drugs':drug_inds[training_indices], 'targets':target_inds[training_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'test', 'setting':'S1', 'index':S1_test_indices, 'drugs':drug_inds[S1_test_indices], 'targets':target_inds[S1_test_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'validation', 'setting':'S1', 'index':S1_validation_indices, 'drugs':drug_inds[S1_validation_indices], 'targets':target_inds[S1_validation_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'test', 'setting':'S2', 'index':S2_test_indices, 'drugs':drug_inds[S2_test_indices], 'targets':target_inds[S2_test_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'validation', 'setting':'S2', 'index':S2_validation_indices, 'drugs':drug_inds[S2_validation_indices], 'targets':target_inds[S2_validation_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'test', 'setting':'S3', 'index':S3_test_indices, 'drugs':drug_inds[S3_test_indices], 'targets':target_inds[S3_test_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'validation', 'setting':'S3', 'index':S3_validation_indices, 'drugs':drug_inds[S3_validation_indices], 'targets':target_inds[S3_validation_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'test', 'setting':'S4', 'index':S4_test_indices, 'drugs':drug_inds[S4_test_indices], 'targets':target_inds[S4_test_indices]}), 
+    pd.DataFrame({'roundID':random_seed, 'subset':'validation', 'setting':'S4', 'index':S4_validation_indices, 'drugs':drug_inds[S4_validation_indices], 'targets':target_inds[S4_validation_indices]})], 
+    axis = 0, sort = False)
+
+    # Combine the lists of test and validation indices for each setting.
+    S1_test_validation = [S1_test_indices, S1_validation_indices]
+    S2_test_validation = [S2_test_indices, S2_validation_indices]
+    S3_test_validation = [S3_test_indices, S3_validation_indices]
+    S4_test_validation = [S4_test_indices, S4_validation_indices]
+    return(df_indices, training_indices, S1_test_validation, S2_test_validation, S3_test_validation, S4_test_validation)
+
+if __name__ == "__main__":
+    # Select a seed or multiple seeds for controlling the randomness in creating the splits. 
+    random_seeds = [2688385916]
+    # Choose the percentage of drugs and targets that are defined to be in area S1 and the percentage of area S1 pairs that will be used as training data.
+    split_percentage = 0.6
+    datasets = ["davis", "metz", "merget"] 
```

Comment:
Previously I also calculated tests for chessboard data. Could you add it also in here (see L0ridge.py file and generate_chess function)? Or have we decided to exclude these tests from the revision version of the article?